### PR TITLE
Add TypeScript config example to README

### PR DESCRIPTION
**Pull request by dekoza**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Add this to your `.pre-commit-config.yaml`:
         rev: ''  # Use the sha / tag you want to point at
         hooks:
         -   id: eslint
+            types: [file]
```
Comment (asottile) [inline README.md:19]: this is not correct -- this will pass all files in your repository to `eslint`

File: README.md [hunk 2]
```diff
@@ -24,9 +25,11 @@ When using plugins with `eslint` you'll need to declare them under
...
             additional_dependencies:
             -   eslint@4.15.0
             -   eslint-config-google@0.7.1
             -   eslint-loader@1.6.1
             -   eslint-plugin-react@6.10.3
             -   babel-eslint@6.1.2
+
```
Comment (asottile) [inline README.md:35]: lol do I need to add pre-commit hooks to this repository too?

File: README.md (hunk 1 above)
Comment (dekoza) [reply README.md:19]: You're obviously right but the default type `javascript` omits TypeScript files (or I'm doing something wrong which is more probable). I'll look for a solution.

File: README.md [hunk 3]
```diff
@@ -30,3 +30,18 @@ When using plugins with `eslint` you'll need to declare them under
...
+By default only `*.js` files are taken into consideration.
+If you want to use eslint on TypeScript codebases you need
+to start from this template:
+
+    -   repo: https://github.com/pre-commit/mirrors-eslint
+        rev: ''  # Use the sha / tag you want to point at
+        hooks:
+        -   id: eslint
+            files: .+\.[jt]sx?$
```
Comment (asottile) [inline README.md:42]: you can leave off the `.+` part -- pre-commit uses `search` over regexes

File: README.md [hunk 4]
```diff
@@ -30,3 +30,18 @@ When using plugins with `eslint` you'll need to declare them under
...
+            types: [file]
+            additional_dependencies:
+            -   eslint@^7.0.0
+            -   other dependencies...
```
Comment (asottile) [inline README.md:46]: I would leave off the `additional_dependencies` bit here -- that's covered in the other section

**Pull request merged**